### PR TITLE
code optimization to remove if and for

### PR DESCRIPTION
**Pull request by MrJithil**
Completed a TODO task. Along with that, optimised the code by removing an if and for

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -417,15 +417,9 @@ module.exports = class Minipass extends Stream {
...
-    // TODO: replace with a spread operator when Node v4 support drops
-    const args = new Array(arguments.length)
-    args[0] = ev
-    args[1] = data
-    if (arguments.length > 2) {
-      for (let i = 2; i < arguments.length; i++) {
-        args[i] = arguments[i]
-      }
-    }
+    const args = new Array(...arguments);
```
Comment (benjamingr) [inline index.js:420]: Nit: this can be:

```js
const args = [...arguments];
``

File: index.js [hunk 2]
```diff
@@ -417,15 +417,9 @@ module.exports = class Minipass extends Stream {
...
-      for (let i = 2; i < arguments.length; i++) {
-        args[i] = arguments[i]
-      }
-    }
+    const args = [...arguments];
+    args[0] = ev;
+    args[1] = data;
 
     try {
       return super.emit.apply(this, args)
```
Comment (isaacs) [inline index.js:425]: Can get rid of all of the above, I think, and just change this one line to:

```
return super.emit(...arguments)
```

That's what the `TODO` item was saying.  `arguments[0]` and `arguments[1]` are already ev and data, respectively.

**Review by isaacs — CHANGES_REQUESTED** [review]
Go a bit further.

**Comment by isaacs** [comment]
Got swept up in some other perf enhancements on main.  Closing.  Thanks :)

**Pull request closed without merging**